### PR TITLE
ICU-22423 Add OpenSSF Scorecard Action

### PR DESCRIPTION
**Pull request by pnacht**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22423
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

As mentioned in the JIRA issue, the Scorecard Action will monitor the project's security posture and populate its Security Dashboard with actionable suggestions.

There is also a badge that can be added to the project's README showing the project's current score: [OpenSSF Scorecard](https://securityscorecards.dev/viewer/?uri=github.com/unicode-org/icu). I haven't added it yet because the README currently only has build status badges and I didn't want to impose. However, if you'd like me to add the badge, just say the word (including where you'd like it to be) and I'll add it to the PR.

### CODE REVIEW

File: .github/workflows/scorecard.yml
Comment (echeran) [inline .github/workflows/scorecard.yml:13]: Looks good. Crontab.guru says [this means “At 15:23 on Saturday.”](https://crontab.guru/#23_15_*_*_6) (once a week)

File: .github/workflows/scorecard.yml
Comment (echeran) [inline .github/workflows/scorecard.yml:32]: Since the [tags in the actions/checkout repo](https://github.com/actions/checkout/tags) include `v3.5.3`, how do you feel about changing the git ref here from the hexadecimal sha id to the tag string?

File: .github/workflows/scorecard.yml
Comment (echeran) [inline .github/workflows/scorecard.yml:37]: ditto question, and below

**Review by echeran — APPROVED** [review]
LGTM modulo an optional minor question. Adding the badge to the root README sounds fine to me. WDYT @markusicu ?

File: .github/workflows/scorecard.yml
Comment (pnacht) [reply .github/workflows/scorecard.yml]: Done. Sorry for the delay!

**Review by echeran — APPROVED** [review]

**Comment by echeran** [comment]
Thanks for your work again, @pnacht! I'll click the merge button on your behalf since it might not be activated for you.

**Comment by FrankYFTang** [comment]
@echeran please address the test breakage of https://github.com/unicode-org/icu/actions/runs/6030584531/job/16362697011

**Comment by pnacht** [comment]
Urgh, sorry about that!

**Comment by echeran** [comment]
> @echeran please address the test breakage of https://github.com/unicode-org/icu/actions/runs/6030584531/job/16362697011

Done in #2571

**Pull request merged**
